mongodb-native/src/main/java/org/ballerinalang/mongodb/RecordIteratorUtils.java

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.mongodb;
+
+import static io.ballerina.runtime.api.utils.StringUtils.fromString;
+
+import com.mongodb.client.MongoCursor;
+
+import org.bson.Document;
+
+import io.ballerina.runtime.api.creators.ErrorCreator;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.internal.JsonParser;
+
+/**
+ * This class provides functionality for the `RecordIterator` to iterate through the MongoCursor.
+ *
+ * @since 3.0.0
+ */
+public class RecordIteratorUtils {
+    public RecordIteratorUtils() {
+    }
+
+    public static Object nextResult(BObject customRecordIterator, BObject recordIterator) {
+        return nextResult(recordIterator);
+    }
+
+    public static Object nextResult(BObject recordIterator) {
+        MongoCursor<Document> results = (MongoCursor<Document>) recordIterator.getNativeData(
+                                         MongoDBConstants.RESULT_SET_NATIVE_DATA_FIELD);
+        if (results.hasNext()) {
+            return JsonParser.parse(results.next().toJson());
```

Comment:
Shouldn't this be a record and not json?

Reply:
To map the record we couldn't get the meta data of MongoDB documents. GetStats method which gave the capability to get the metadata is now seems deprecated. https://jira.mongodb.org/browse/CSHARP-1930?focusedCommentId=1515262&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-1515262